server/aesdsocket.c

```diff
@@ -0,0 +1,346 @@
+
+
+        char * write_buf;
+        write_buf = (char *)malloc(sizeof(char) * MAXDATASIZE);
+
+	write_file_fd = open(FILE_PATH, O_RDWR | O_APPEND | O_CREAT, S_IRWXU);
+
+        if (write_file_fd == -1)
+        {
+        printf("2\n");
+            syslog(LOG_ERR, "Error in opening file");
+            close(socket_fd);
+            closelog();
+            free(buf);
+            free(write_buf);
+            return -1;
+        }
+
+
+
+
+    while(1)
+    {
+
+	client_fd = accept(socket_fd, (struct sockaddr *)&their_addr, &addr_size);
+
+	if(client_fd == -1)
+	{
+		printf("1\n");
+	    syslog(LOG_ERR, "Error in retrieving client fd\n");
+            close(socket_fd);
+            closelog();
+	    return -1;
+	}
+
+	inet_ntop(AF_INET, get_in_addr((struct sockaddr *)&their_addr), ipstr, sizeof ipstr);
+        syslog(LOG_DEBUG,"Accepted connection from %s", ipstr);
+	int numbytes = 0;
+/*
+    	char *buf;
+
+    	buf = (char *)malloc(sizeof(char) * MAXDATASIZE);
+
+
+        char * write_buf;
+        write_buf = (char *)malloc(sizeof(char) * MAXDATASIZE);
+
+*/
+
+
+/*    	write_file_fd = open(FILE_PATH, O_RDWR | O_APPEND | O_CREAT, S_IRWXU);
+
+    	if (write_file_fd == -1)
+    	{
+	printf("2\n");
+            syslog(LOG_ERR, "Error in opening file");
+            close(socket_fd);
+            closelog();
+	    free(buf);
+	    free(write_buf);
+	    return -1;
+    	}*/
+
+	int buf_location = 0;
+
+	do //while((numbytes = recv(client_fd, buf + buf_location, MAXDATASIZE -1, 0)) > 0)
+	{
+	    numbytes = recv(client_fd, buf + buf_location, MAXDATASIZE, 0);
+	    if(numbytes == -1) {
+printf("3\n");
+        	perror("recv");
+		close(socket_fd);
+		closelog();
+		free(buf);
+		free(write_buf);
+        	return -1;
+    	    }
+	    ++realloc_count;
+	    buf_location += numbytes;
+            check_tot +=numbytes;
+	    //char * temp_buf;
+            buf = (char *)realloc(buf, realloc_count*MAXDATASIZE*(sizeof(char)));
+	    if(buf == NULL)
+	    {
+printf("4\n");
+            	syslog(LOG_ERR, "Error in realloc");
+            	close(socket_fd);
+            	closelog();
+            	free(buf);
+		free(write_buf);
+            	return -1;
+       	     }
+//	     buf = temp_buf;
+
+
+	//	if(strchr(buf, '\n') != NULL) break;
+	} while(strchr(buf, '\n') == NULL);
+	buf[buf_location] = '\0';
+	//buf[check_tot] = '\0';
+
+        int wr;
+        wr = write(write_file_fd, buf, buf_location);
+        if(wr == -1)
+        {
+printf("5\n");
+            syslog(LOG_ERR, "Error in writing to file");
+            close(socket_fd);
+            closelog();
+            free(buf);
+	    free(write_buf);
+            return -1;
+        }
+
+	// Set cursor to beginning of file for reading
+	//lseek(write_file_fd, 0, SEEK_SET);
+
+	// buffer used for reading from file and sending through socket
+	//int send_bytes_check = 0;
+	//char * temp_write_buf;
+	write_buf = (char *) realloc(write_buf, check_tot*(sizeof(char)));
```

Comment:
@Sharath-Jonnala what assumption doet his make about the total size of received packets relative to the size of the heap?